model/src/main/kotlin/utils/FindingsMatcher.kt

```diff
@@ -70,11 +57,8 @@ class FindingsMatcher(
     /**
      * Get the licenses found in all commonly named license files, if any, or an empty list otherwise.
      */
-    private fun getRootLicenses(licenseFindings: Collection<LicenseFinding>): List<String> =
-        licenseFindings
-            .filter { licenseFileMatcher.matches(it.location.path) }
-            .map { it.license }
-            .distinct()
+    private fun getRootLicenseFindings(licenseFindings: Set<LicenseFinding>): Set<LicenseFinding> =
```

Comment:
Nit: The `licenseFindings` input could stay to be a `Collection`.

Reply:
Could yes, but should it? It's a private function that's only ever called with sets as arguments, making it more generic wouldn't gain anything.